src/universal/components/NewMeeting.js

```diff
@@ -138,37 +141,29 @@ class NewMeeting extends Component<Props> {
     const {meetingType, viewer} = this.props;
     const {team} = viewer;
     const {newMeeting, teamName} = team;
-    const {facilitatorStageId, localStage} = newMeeting || {};
-    const meetingSlug = meetingTypeToSlug[meetingType];
+    const {facilitatorStageId, localPhase, localStage} = newMeeting || {};
     const meetingLabel = meetingTypeToLabel[meetingType];
     const inSync = localStage ? localStage.localStageId === facilitatorStageId : true;
+    const localPhaseType = localPhase && localPhase.phaseType;
     return (
       <MeetingContainer>
         <Helmet title={`${meetingLabel} Meeting for ${teamName} | Parabol`} />
         <NewMeetingSidebar gotoStageId={this.gotoStageId} meetingType={meetingType} viewer={viewer} />
         <MeetingArea>
           <MeetingAreaHeader>
-            <NewMeetingPhaseHeading />
+            <NewMeetingPhaseHeading meeting={newMeeting} />
             <NewMeetingAvatarGroup
               gotoStageId={this.gotoStageId}
               team={team}
             />
           </MeetingAreaHeader>
           <ErrorBoundary>
-            <Switch>
-              <Route
-                path={`/${meetingSlug}/:teamId/${phaseTypeToSlug[CHECKIN]}`}
-                render={() => <NewMeetingCheckIn gotoNext={this.gotoNext} meetingType={meetingType} team={team} />}
-              />
-              <Route
-                path={`/${meetingSlug}/:teamId/${phaseTypeToSlug[REFLECT]}`}
-                render={() => <RetroReflectPhase team={team} />}
-              />
-              <Route
-                path={`/${meetingSlug}/:teamId`}
-                render={() => <NewMeetingLobby meetingType={meetingType} team={team} />}
-              />
-            </Switch>
+            <React.Fragment>
+              {localPhaseType === CHECKIN && <NewMeetingCheckIn gotoNext={this.gotoNext} meetingType={meetingType} team={team} />}
```

Comment:
can't rely on the router inside the meeting because the router updates async from the route in the address bar.